sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt

```diff
@@ -0,0 +1,57 @@
+package io.sentry.android.navigation
+
+import android.os.Bundle
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SentryLevel.INFO
+import io.sentry.TypeCheckHint
+import java.lang.ref.WeakReference
+
+class SentryNavigationListener(
+    private val hub: IHub = HubAdapter.getInstance()
+) : NavController.OnDestinationChangedListener {
+
+    private var fromRef: WeakReference<NavDestination>? = null
+    private var fromArgs: Bundle? = null
+
+    override fun onDestinationChanged(
+        controller: NavController,
+        destination: NavDestination,
+        arguments: Bundle?
+    ) {
+        addBreadcrumb(destination, arguments)
+        fromRef = WeakReference(destination)
+        fromArgs = arguments
+    }
+
+    private fun addBreadcrumb(destination: NavDestination, arguments: Bundle?) {
+        val breadcrumb = Breadcrumb().apply {
+            type = "navigation"
+            category = "navigation"
+
+            val from = fromRef?.get()?.route
+            from?.let { data["from"] = it }
+            fromArgs?.let { args ->
+                data["from_arguments"] = args.keySet().filter {
+                    it != NavController.KEY_DEEP_LINK_INTENT // there's a lot of unrelated stuff
+                }.associateWith { args[it] }
+            }
+            val to = destination.route
+            to?.let { data["to"] = it }
+            // TODO: should this be hidden behind Pii flag?
```

Comment:
I guess we should do it, because the user may pass username + password between composables/screens and this might be a problem?

Reply:
Better to be safe than sorry. People could pass random PII things as arguments. This also applies for the other arguments.

Reply:
We don't use `defaultPii` for such things, not on Flutter nor React Native, the reason is to reproduce issues you need this data and it's not related to PII (most likely), if there's PII and they don't want to, they can strip out using `beforeBreadcrumb`, or `beforeSend` or even on the server-side, maybe not even using it at all.

Reply:
> We don't use defaultPii for such things, not on Flutter nor React Native, the reason is to reproduce issues you need this data and it's not related to PII (most likely), if there's PII and they don't want to, they can strip out using beforeBreadcrumb, or beforeSend or even on the server-side, maybe not even using it at all.

Agreed, but this has the assumption it's unlikely to have PII. Roman mentioned there could be login/password. 
If so, I'd suggest adding behind `sendDefaultPii`. We haven't used much for the lack of use cases, but the feature is there.